### PR TITLE
[9.0][website_portal_v10] Backport of website_portal from v10

### PR DESCRIPTION
**Pull request by yajo**
tl;dr: `website_portal` v9 is evil, v10 is good; backporting.

After initiating some work with `website_portal` v9 core version, it lacks some patches that landed in its v8 OCA backport that almost render the module unusable.

Well, it's usable, but layout is awful and unchangeable. Presuming https://github.com/odoo/odoo/pull/12225 will never get merged, and if it does, future migrations to v10 `website_portal` would require a big amount of work due to the fact that current implementation has to be heavily workarounded to fix that bug and the v10 one is just good, I'm backporting the new version of this module to allow work to start now with its base.

The backport renames the module to avoid collisions with existing installations, although you should be warned that both are not compatible.

@Tecnativa

### CODE REVIEW

File: website_portal_v10/README.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
+.. image:: https://img.shields.io/badge/licence-LGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/lgpl-3.0-standalone.html
+   :alt: License: LGPL-3
+
+====================================
+Website Portal (Backported From v10)
+====================================
+
+This module replaces the functionality of ``website_portal`` to support sane
+layout and extensibility and allow you to have a special layout just for portal
```
Comment (pedrobaeza) [inline website_portal_v10/README.rst]: s/and allow/and to allow

File: website_portal_v10/README.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+pages, have custom controllers for pages, have a menu indicating all sections
```
Comment (pedrobaeza) [inline website_portal_v10/README.rst]: s/, have a/, to have a

Comment (pedrobaeza) [inline website_portal_v10/README.rst]: 2 times

File: website_portal_v10/README.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+for the portal features, inherit everything without problems, and prepare for
```
Comment (pedrobaeza) [inline website_portal_v10/README.rst]: s/, inherit/, to inherit

Comment (pedrobaeza) [inline website_portal_v10/README.rst]: s/, and prepare/, and to prepare

File: website_portal_v10/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+    'name': 'Website Portal (Backported From v10)',
+    'category': 'Website',
+    'summary': 'Account Management Frontend for your Customers',
+    'version': '9.0.1.0.0',
+    'description': """
```
Comment (pedrobaeza) [inline website_portal_v10/__openerp__.py]: Remove description key, as we have README.rst

File: website_portal_v10/controllers/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+import main
```
Comment (pedrobaeza) [inline website_portal_v10/controllers/__init__.py]: `from . import main`

File: website_portal_v10/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+from openerp import http
+from openerp.http import request
+from openerp import tools
+from openerp.tools.translate import _
+
+from openerp.fields import Date
+
+
+class website_account(http.Controller):
```
Comment (pedrobaeza) [inline website_portal_v10/controllers/main.py]: Although a backport, we should follow OCA guidelines. Merge won't be possible anyway because there are a lot of things that diverges soon (that were the conclusions when backporting account_bank_statement_import module).

File: website_portal_v10/controllers/main.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+    _items_per_page = 10
```
Comment (pedrobaeza) [inline website_portal_v10/controllers/main.py:13]: Maybe this can be put as ir.parameter.

File: website_portal_v10/__openerp__.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (elicoidal) [inline website_portal_v10/__openerp__.py:5]: Can you add copyright notices in all the files?

File: website_portal_v10/controllers/main.py (hunk 7 above)
Comment (yajo) [inline website_portal_v10/controllers/main.py:13]: That should go to odoo master.

Comment (pedrobaeza) [inline website_portal_v10/controllers/main.py:13]: Yeah, sure

File: website_portal_v10/__openerp__.py (hunk 8 above)
Comment (yajo) [inline website_portal_v10/__openerp__.py:5]: Fixed

**Comment by pedrobaeza** [comment]
:+1:

**Comment by rafaelbn** [comment]
Tested in runbot :+1: Thanks!

**Comment by lasley** [comment]
Excellent work 👍

**Comment by yajo** [comment]
Hi bots, 

This is a backport, that's why some users are not in OCA and no tests cover the code.

I hope you understand.

**Comment by rafaelbn** [comment]
Please check travis errors

**Comment by yajo** [comment]
Bots are :green_heart: now. :blush:

**Comment by lasley** [comment]
Functional + code 👍

**Comment by rafaelbn** [comment]
This PR cannot be merged until https://github.com/OCA/website/pull/204#issuecomment-224326596 is solved. cc @pedrobaeza

**Comment by lasley** [comment]
Good call @rafaelbn , the line about not being included in CLA is required in the ReadMe 

```
**This module is a backport from Odoo SA and as such, it is not included in the OCA CLA. That means we do not have a copy of the copyright on it like all other OCA modules.**
```

File: website_portal_v10/README.rst [hunk 9]
```diff
@@ -82,6 +82,10 @@ welcomed feedback.
 Credits
 =======
 
+**This module is a backport from Odoo SA and as such, it is not included in the
+OCA CLA. That means we do not have a copy of the copyright on it like all
+other OCA modules.**
+
```
Comment (rafaelbn) [inline website_portal_v10/README.rst:88]: Dear @OCA/board is this OK? Could we merge? I'm not sure in this point. https://github.com/OCA/website/pull/204#issuecomment-232560496

Comment (lasley) [inline website_portal_v10/README.rst:88]: I'm not board- but in the Contributing guidelines in the Backporting section, this is stated - https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#backporting-odoo-modules

**Comment by rafaelbn** [comment]
Thanks @lasley .
